tinygrad/codegen/linearizer.py

```diff
@@ -314,10 +314,7 @@ def linearize(self):
         fake_global_idxs = [x*0 for x in global_idxs]
         self.global_store(-1, fake_global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs, acc)  # store accumulators
         self.uop(UOps.BARRIER, None, [], ())
-        self.uop(UOps.ENDLOOP, None, [], (loop_local_idxs, "local"))
-
-        # local indexs are over, 0 them out
-        local_idxs = [x*0 for x in local_idxs]
+        self.uop(UOps.ENDLOOP, None, [], ([loop_local_idxs[i]*0 if i < len(self.group_for_reduce) else loop_local_idxs[i] for i in range(0,self.local_dims+len(self.group_for_reduce))], "local"))
```

Comment:
`loop_local_idxs[i]*0`? Is that not just `0`?

Reply:
These have to be `Variable`s.

Reply:
If they have to be variables, you could use `Variable.num(0)`